app/src/main/java/org/p2p/wallet/newsend/model/SendSolanaFee.kt

```diff
@@ -161,18 +163,39 @@ data class SendSolanaFee constructor(
         val shouldTryReduceAmount = isAllowedToCorrectAmount && !isSourceSol && !isEnoughSolBalance
         val hasAlternativeFeePayerTokens = alternativeFeePayerTokens.isNotEmpty()
         val isValidToSwitchOnSource = supportedFeePayerTokens?.contains(sourceToken) ?: true
+        // if there is enough SPL token balance to cover amount and fee
+        val shouldSwitchToSpl =
+            !isSourceSol &&
+                sourceTokenTotal.isMoreThan(totalNeeded) &&
+                isValidToSwitchOnSource
+
+        Timber.i(buildString {
+            appendLine("isSourceSol = $isSourceSol")
+            appendLine("isAllowedToCorrectAmount = $isAllowedToCorrectAmount")
+            appendLine("totalNeeded = $totalNeeded")
+            appendLine("isEnoughSolBalance = $isEnoughSolBalance")
+            appendLine("shouldTryReduceAmount = $shouldTryReduceAmount")
+            appendLine("hasAlternativeFeePayerTokens = $hasAlternativeFeePayerTokens")
+            appendLine("alternativeFeePayerTokens = ${alternativeFeePayerTokens.map(Token.Active::tokenSymbol)}")
+            appendLine("isValidToSwitchOnSource = $isValidToSwitchOnSource")
+            appendLine("shouldSwitchToSpl = $shouldSwitchToSpl")
+        })
         return when {
-            // if there is enough SPL token balance to cover amount and fee
-            !isSourceSol && sourceTokenTotal.isMoreThan(totalNeeded) &&
-                isValidToSwitchOnSource -> FeePayerState.SwitchToSpl(sourceToken)
-            hasAlternativeFeePayerTokens -> FeePayerState.SwitchToSpl(alternativeFeePayerTokens.first())
+            shouldSwitchToSpl -> {
```

Comment:
the main problem is here
User selected AVA token to send to account a.p2p.sol
a.p2p.sol doesn't have token account so user needs to pay fee to create one
we switch fee payer token to SOL
after returning to main send screen we see that fee is calculated in SOL
but when tapping on SWITCH_MODE button, fee_payer token is switched to USDC 

in this case I saw that 
isEnoughSolBalance = true
hasAlternativeFeePayerTokens = true
and that's why in when case SendSolanaFee selects new fee payer token = USDC